File: TTK4255 - Robotsyn/Øving 6/python/test_normalize_points.py
import numpy as np

def test_normalize_points(pts):
    mean = np.mean(pts, axis=0)
    dist = np.mean(np.linalg.norm(pts - mean, axis=1))
    if np.absolute(mean[0]) > 0.01 or np.absolute(mean[1]) > 0.01:
        print('Translation is NOT GOOD, centroid was (%g, %g), should be (0,0).' % (mean[0], mean[1]))
        return
    else:
        print('Translation is GOOD.')
    if np.absolute(dist - np.sqrt(2)) > 0.1:
        print('Scaling is NOT GOOD, mean distance was %g, should be sqrt(2).' % dist)
        return
    else:
        print('Scaling is GOOD.')

File: TTK4255 - Robotsyn/Øving 6/matlab/test_normalize_points.m
function test_normalize_points(pts_n)
    centroid = mean(pts_n);
    dist = mean(vecnorm(pts_n, 2, 2));
    if abs(centroid(1)) > 0.01 || abs(centroid(2)) > 0.01
        fprintf('Translation is NOT GOOD: Centroid was (%g, %g), should be (0, 0).\n', centroid(1), centroid(2));
    else
        fprintf('Translation is GOOD.\n');
    end
    if abs(dist - sqrt(2)) > 0.1
        fprintf('Scaling is NOT GOOD: Mean distance was %g, should be sqrt(2).\n', dist)
    else
        fprintf('Scaling is GOOD.\n');
    end
end